src/espIdf/serial/serialPort.ts

```diff
@@ -87,24 +86,50 @@ export class SerialPort {
   private list(workspaceFolder: vscode.Uri): Thenable<SerialPortDetails[]> {
     return new Promise(async (resolve, reject) => {
       try {
+        const listOfSerialPorts = await SerialPortLib.SerialPort.list();
+
+        const choices = listOfSerialPorts.map((item) => {
+          return new SerialPortDetails(
+            item.path,
+            item.manufacturer,
+            item.vendorId,
+            item.productId
+          );
+        });
         const pythonBinPath = idfConf.readParameter(
           "idf.pythonBinPath",
           workspaceFolder
         ) as string;
-        const buff = await spawn(pythonBinPath, ["get_serial_list.py"]);
-        const regexp = /\'(.*?)\'/g;
-        const arrayPrint = buff.toString().match(regexp);
-        const choices: SerialPortDetails[] = Array<SerialPortDetails>();
+        const idfPath = idfConf.readParameter(
+          "idf.espIdfPath",
+          workspaceFolder
+        );
+        const esptoolPath = join(
+          idfPath,
+          "components",
+          "esptool_py",
+          "esptool",
+          "esptool.py"
+        );
+        async function processPorts(serialPort: SerialPortDetails) {
```

Comment:
Could we add validation that esptool.py exists ?

Reply:
ok, added